app/hooks/useDaiBalance.tsx

```diff
@@ -32,7 +32,7 @@ export default function useDaiBalance() {
     }
   );
 
-  if (!balance || error) {
+  if (!balance || error !== undefined) {
```

Comment:
This should be equivalent I believe :thinking:

Reply:
Maybe just testing for `error` would be clearer?

```suggestion
  if (error) {
```

Reply:
Just testing for `error` was not enough, it seems that the error can be triggered, but the balance still not set. 
But I removed the check for `undefined`.